spec/oauth2/access_token_spec.rb

```diff
@@ -75,6 +75,41 @@ def assert_initialized_token(target) # rubocop:disable Metrics/AbcSize
       assert_initialized_token(target)
       expect(target.expires_at).to be_a(Integer)
     end
+
+    describe 'expires_latency' do
+      let(:expires_at) { 1_530_000_000 }
+      let(:expires_in) { 100 }
+      let(:expires_latency) { 10 }
+      let(:hash) do
+        {
+          :access_token => token,
+          :expires_latency => expires_latency,
+          :expires_in => expires_in,
+        }
+      end
+
+      it 'sets it via options' do
+        target = described_class.from_hash(client, hash.merge(:expires_latency => expires_latency.to_s))
+        expect(target.expires_latency).to eq 10
+      end
+
+      it 'sets it nil by default' do
+        hash.delete(:expires_latency)
+        target = described_class.from_hash(client, hash)
+        expect(target.expires_latency).to be_nil
+      end
+
+      it 'reduces expires_at by the given amount' do
+        allow(Time).to receive(:now).and_return(expires_at)
+        target = described_class.from_hash(client, hash)
+        expect(target.expires_at).to eq(expires_at + expires_in - expires_latency)
+      end
+
+      it 'reduces expires_at by the given amount if expires_at is provided as option' do
```

Comment:
Perhaps one more additional test - `reduces expires_at by the given amount if expires_at is not provided as option`?

Reply:
I think `reduces expires_at by the given amount` covers this case since `expires_at` is not provided by default and this test explicitly says that it is provided, and adds it to the hash.